### PR TITLE
Sparse trajectories only

### PR DESCRIPTION
**Pull request by gilesknap**
This is OK to merge into master.

It implements the sparse PVT points when 'start of row' triggering is requested. It only works for linear rows but does not break non-linear rows (it just creates the same PVT points as before the change).

This branch is now separate from velocity-mode-changes and so it still works with the current version of the trajectory PROG.

### CODE REVIEW

File: malcolm/modules/pmac/parts/pmacchildpart.py [hunk 1]
```diff
@@ -570,3 +622,17 @@ def insert_gap(self, point, next_point, completed_steps):
...
         self.profile["userPrograms"][-1] = user_program
+
+    def is_equidistant(self, p1, p2):
+        result = False
+        # handle p1 = None, i.e. no previous point
+        if p2.duration == p2.duration:
+            result = True
+            for axis_name, motor_info in self.axis_mapping.items():
+                if not np.isclose(
+                    p1.lower[axis_name] - p1.positions[axis_name],
```
Comment (thomascobb) [inline malcolm/modules/pmac/parts/pmacchildpart.py]: Should we alos check upper-positions?

File: malcolm/modules/pmac/parts/pmacchildpart.py [hunk 2]
```diff
@@ -570,3 +622,17 @@ def insert_gap(self, point, next_point, completed_steps):
         self.profile["velocityMode"][-1] = CURRENT_TO_NEXT
         user_program = self.get_user_program(PointType.START_OF_ROW)
         self.profile["userPrograms"][-1] = user_program
+
+    def is_equidistant(self, p1, p2):
```
Comment (thomascobb) [inline malcolm/modules/pmac/parts/pmacchildpart.py]: should this be is_same_velocity?

File: malcolm/modules/pmac/parts/pmacchildpart.py (hunk 1 above)
Comment (gilesknap) [reply malcolm/modules/pmac/parts/pmacchildpart.py]: yep

File: malcolm/modules/pmac/parts/pmacchildpart.py (hunk 2 above)
Comment (gilesknap) [reply malcolm/modules/pmac/parts/pmacchildpart.py]: yes

**Pull request merged**